### DISCUSSION TITLE
Reload after updating environment variable?

### DISCUSSION THREAD

**Discussion by Jojoshua** [vercel/vercel#7728] [Help] [answered]
I have a SvelteKit app deployed and I am using several environment variables. When I update a variable it is not reflected in the application until I make a new source code change and force rebuild/deploy of the app. How can I enforce updating these variables without redeploying?

**Comment by leerob** [comment, marked as answer]
Correct, you will need a new deployment to pick up the changed environment variables during the build. 

Alternatively, you could fetch dynamic values at runtime using Page Endpoints or an API route inside Svelte. Could even fetch from the Vercel API if you wanted.

https://vercel.com/docs/rest-api#endpoints/projects/retrieve-the-environment-variables-of-a-project-by-id-or-name

**Comment by sztadii** [comment]
@leerob I created the file in /pages/api/status.ts with the below content.
```
import type { NextApiRequest, NextApiResponse } from 'next'

export default async function getApiStatus(req: NextApiRequest, res: NextApiResponse) {
  const result = {
    isDashboardUnderMaintenance: process.env.DASHBOARD_UNDER_MAINTENENCE === 'true',
  }
  res.status(200).json(result)
}
```
This file is creating for me an endpoint where I wanted to return the latest env variables.
Unfortunately, I need to rebuild the project to get the latest env variables.
After reading your message I understand that I do not need to do that. 
Then why is it happening?

**Reply by amyegan** [reply]
Hi @sztadii! This is expected behavior. You would need to use the [REST API](https://vercel.com/docs/rest-api#endpoints/projects/retrieve-the-environment-variables-of-a-project-by-id-or-name) instead of `process.env.DASHBOARD_UNDER_MAINTENANCE` to get the latest environment variables without redeploying.

**Reply by sztadii** [reply]
Thanks for the quick feedback :)
